docs/test-suite/writing-portable-tests.md

```diff
@@ -0,0 +1,526 @@
+# Writing portable tests
+
+This page is a tutorial on how to write a new test for the [EESSI test suite](https://github.com/EESSI/test-suite).
+
+If you already know how to write regular ReFrame tests, we suggest you read the [High-level overview](#high-level-overview) and [Test requirements](#test-requirements) sections, then skip ahead to [Step 3: implementing as a portable ReFrame test](#as-portable-reframe-test).
+
+## High-level overview
+
+In this tutorial, you will learn how to write a test for the [EESSI test suite](https://www.eessi.io/docs/test-suite/). It is important to realize in which context the test suite will be run. Roughly speaking, there are three uses:
+
+- Running tests for one (or a few) particular applications, as part of the [workflow of adding new software to EESSI](https://www.eessi.io/docs/adding_software/contribution_policy/#testing),  to validate the sanity of the (new) installation
+- Regular (e.g. daily) runs, on a set of HPC clusters, to identify performance regressions
+- By an end-user of EESSI, who runs either a specific test or the full test suite, to validate the functionality of EESSI (or a particular software in EESSI) on the end-user's system
+
+The test suite will contain a combination of real-life use cases for end-user scientific software (e.g. tests for GROMACS, TensorFlow, CP2K, OpenFOAM, etc) and low level tests (e.g. OSU Microbenchmarks).
+
+The tests in the EESSI test suite are developed using the [ReFrame HPC testing framework](https://reframe-hpc.readthedocs.io/en/stable/). Typically, ReFrame tests hardcode system specific information (core counts, performance references, etc) in the test definition. The EESSI test suite aims to be portable, and implements a series of standard [hooks](#REFERENCE_TO_HOOKS_API_DOCS) to replace information that is typically hardcoded. All system-specific information is then limited to the ReFrame configuration file. As an example: rather than hardcoding that a test should run with 128 tasks (i.e. because a system has 128 core nodes), the EESSI test suite has a hook that can define a test should be run on a "single, full node". The hook queries the ReFrame configuration file for the amount of cores per node, and specifies this number as the corresponding amount of tasks. Thus, on a 64-core node, this test would run with 64 tasks, while on a 128-core node, it would run 128 tasks.
+
+## Test requirements
+
+To be useful in the aforementioned scenarios, tests need to satisfy a number of requirements. 
+
+- Tests are implemented in the [ReFrame HPC testing framework](https://reframe-hpc.readthedocs.io/en/stable/).
+- Multiple test cases may be implemented for a single software package.
+- Tests should run in a reasonable amount of time (less than 1 hour) for all the scales for which it is defined to be valid.
+- There should be at least one light test case that can be run in less then ~5 minutes. This test should be marked with the 'CI' tag.
```

Comment:
Hmmmm, yes, although on a laptop, it might be too ambitious to require less than 5 minute runtime. I guess implicitely I mean: it should be 5 minutes on a `1_node` scale, given a decently-sized machine (16 cores or so, recent generation). I should probably make that _explicit_, but need to think how to do that in a short sentence :D

I like the addition of low-memory as well. I guess we should just say "on a modern day desktop / reasonably sized VM (in 2024 that means ~16 cores, 16 GB memory or so)."